csharp/src/Ice/UriParser.cs

```diff
@@ -299,6 +299,22 @@ internal static void RegisterCommon()
                         throw new FormatException("the URI format does not support protocol ice1");
                     }
                 }
+                else if (endpointOptions == null && name == "relative")
```

Comment:
Why is the `endpointOptions` check required here?

Reply:
relative=true|false is only valid for an ice URI, not for an ice+transport URI, since relative proxies are indirect proxies.